### PR TITLE
Align with Governance audit

### PR DESCRIPTION
**Pull request by Ranchhand87**
# Description

Edits to CODE-OF-CONDUCT.md 
Text edit in SECURITY.md
Adding Issue Templates

Please review changes carefully. Please ensure Discussions are enabled prior to merging. Contact @Ranchhand87 if assistance in needed to enable Discussions. 

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated
- [ ] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages.
      **_New section is never added with the code changes._** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [ ] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [ ] The version bounds in `.cabal` files for all affected packages are updated.
      **_If you change the bounds in a cabal file, that package itself must have a version increase._** (See [RELEASING.md](https://github.com/intersectmbo/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/intersectmbo/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

File: .github/ISSUE TEMPLATE/config.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+blank_issues_enabled: true
+contact_links:
+  - name: Contributing guidelines
+    url: https://github.com/cardano-ledger/blob/master/CONTRIBUTING.md
```
Comment (lehins) [inline .github/ISSUE TEMPLATE/config.yml]:
```suggestion
    url: https://github.com/IntersectMBO/cardano-ledger/blob/master/CONTRIBUTING.md
```

File: .github/ISSUE TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    about: Some rules & processes we honor.
+
+  - name: Feature ideas
+    url: https://github.com/cardano-ledger/discussions/categories/ideas
```
Comment (lehins) [inline .github/ISSUE TEMPLATE/config.yml]:
```suggestion
    url: https://github.com/IntersectMBO/cardano-ledger/discussions/categories/ideas
```

File: .github/ISSUE TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    about: Maybe someone else had the same or a similar idea already?
+
+  - name: All issues
+    url: https://github.com/cardano-ledger/issues
```
Comment (lehins) [inline .github/ISSUE TEMPLATE/config.yml]:
```suggestion
    url: https://github.com/IntersectMBO/cardano-ledger/issues
```

File: .github/ISSUE TEMPLATE/feature_idea_ledger.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+description: Idea or request for some feature on the cardano-ledger repository
+labels: [':thought_balloon: idea']
+body:
+  - type: markdown
+    attributes:
+      value: |
+        **Thank you for contributing to our project!** :green_heart:
+
+        Instead of opening this issue, consider [starting a new idea discussion](https://github.com/IntersectMBO/cardano-ledger/discussions/new?category=ideas).
+        That way, we can discuss & refine your idea together, before we adopt it as a feature into the roadmap.
```
Comment (lehins) [inline .github/ISSUE TEMPLATE/feature_idea_ledger.yml]: Why are we starting to use these "ideas"? Don't we already have a CIP process for that?

I am not really opposing this, but I do want to know when we should use one vs the other. We need to clarify this in the template as well.

File: .github/ISSUE TEMPLATE/feature_idea_ledger.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      required: true
+
+  - type: textarea
+    id: what
+    attributes:
+      label: What
+      description: What is this feature roughly about
+      placeholder: |
+        For example describe a new API endpoint, a change in messaging formats,
+        a new configuration option, ...
```
Comment (lehins) [inline .github/ISSUE TEMPLATE/feature_idea_ledger.yml]: We don't have messaging, API endpoints or configuration options.

File: .github/ISSUE TEMPLATE/feature_idea_ledger.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      required: true
+
+  - type: textarea
+    id: how
+    attributes:
+      label: How
+      description: How could we realize this feature
+      placeholder: |
+        Which technical solutions, libraries or systems should be used, which
+        components need to change, steps how to implement this, ...
```
Comment (lehins) [inline .github/ISSUE TEMPLATE/feature_idea_ledger.yml]: It is unlikely that people who are not involved in ledger codebase maintenance will know how a problem has to be solved from the technical point of view. If an issue is created by ledger maintainers then we already know what needs to be written in the ticket
```suggestion
```

File: .github/ISSUE TEMPLATE/feature_idea_ledger.yml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+        Instead of opening this issue, consider [starting a new idea discussion](https://github.com/IntersectMBO/cardano-ledger/discussions/new?category=ideas).
+        That way, we can discuss & refine your idea together, before we adopt it as a feature into the roadmap.
+
+  - type: textarea
+    id: why
+    attributes:
+      label: Why
+      description: Why do we need or want this feature
+      placeholder: |
+        Give context and describe the problem, challenge or opportunity you see
```
Comment (lehins) [inline .github/ISSUE TEMPLATE/feature_idea_ledger.yml]: I think this is the only sensible section for the Ledger project in this template. I would suggest removing all others.

File: CODE-OF-CONDUCT.md [hunk 8]
```diff
@@ -1 +1,128 @@
...
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+.
```
Comment (lehins) [inline CODE-OF-CONDUCT.md]: Dot is on a newline. Also feels like `at` is out of place and the sentence is not fully finished "responsible for enforcement ... " of what?
```suggestion
reported to the community leaders responsible for enforcement at.

```

**Review by lehins — CHANGES_REQUESTED** [review]
There are some issues with an issue template. As I was saying earlier, we don't really need an issue template for the Ledger repo. But, if you insist on us having it, then we either need totally different sections or current suctions need to be removed.

It is not a simple the "what" and a "how" in ledger. There are a lot more nuances, that only Ledger maintainers can answerr: eg. does it require a hard fork, if so can it be changed with an intra-era hard fork, does it require changes to CDDL, serialization of ledger state, ...

So, my question to you is this template targets ledger maintainers or downstream users that request features and report bugs?

File: .github/ISSUE TEMPLATE/feature_idea_ledger.yml (hunk 7 above)
Comment (Ranchhand87) [reply .github/ISSUE TEMPLATE/feature_idea_ledger.yml]: @lehins Committed suggestions. Rewrote Issue temp to be more general for the time being, at least at a place where you and the team can add to it in the future when there is time. Also removed Discussions from the Config. We can revisit that topic separately and consider re-adding at a later date. Hoping this helps.

**Review by lehins — APPROVED** [review]
Looks good. Thank you!

File: .github/ISSUE TEMPLATE/config.yml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+blank_issues_enabled: true
+contact_links:
+  - name: Contributing guidelines
+    url: https://github.com/IntersectMBO/cardano-ledger/blob/master/CONTRIBUTING.md
+    about: Some rules & processes we honor.
+
+  - name: All issues
+    url: https://github.com/IntersectMBO/cardano-ledger/issues
+    about: Check whether your issue is not already covered here.
```
Comment (lehins) [inline .github/ISSUE TEMPLATE/config.yml]:
```suggestion
    about: Check whether your issue is not already covered here.

```

File: .github/ISSUE TEMPLATE/feature_idea_ledger.yml [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+### Environment (if applicable)
+- Cardano Ledger Version:
+- Operating System:
+- Nix/Cabal Version:
+
+### Additional Context
+Add any other context about the problem or request here. Include possible solutions, discussions, or relevant documentation links.
+
+### Labels
+(For maintainers: add relevant labels such as `bug`, `enhancement`, `technical-debt`, etc.)
```
Comment (lehins) [inline .github/ISSUE TEMPLATE/feature_idea_ledger.yml]:
```suggestion
(For maintainers: add relevant labels such as `bug`, `enhancement`, `technical-debt`, etc.)

```

**Review by lehins — APPROVED** [review]

**Pull request merged**